frontends/parsers/p4/p4parser.ypp

```diff
@@ -316,8 +374,121 @@ annotations
 
 annotation
     : "@" name                        { $$ = new IR::Annotation(@1, *$2, {}); }
-    | "@" name "(" expressionList ")" { $$ = new IR::Annotation(@1, *$2, *$4); }
-    | "@" name "(" kvList ")"         { $$ = new IR::Annotation(@1, *$2, *$4); }
+    | "@" name "(" annotationBody ")" { $$ = new IR::Annotation(@1, *$2, *$4); }
+    | PRAGMA name annotationBody END_PRAGMA
```

Comment:
This is a non-standard language feature. Perhaps it should be marked as "experimental".
I thought pragmas were a p4-14 thing only.

Reply:
So we can add a comment saying that this is not a standard P4-16 grammar construct.
We marked these as // experimental in the past.
We could consider standardizing `@pragma` in the spec too.

Reply:
Done in 9667864. (Pragmas appear to be non-standard in P4₁₄ too!)